sim/src/image.rs

```diff
@@ -852,6 +873,50 @@ impl Images {
 
         (flash, resets)
     }
+
+    /// Verify the image in the given flash device, the specified slot
+    /// against the expected image.
+    fn verify_images(&self, flash: &SimMultiFlash, slot: usize, against: usize) -> bool {
```

Comment:
Maybe would be clearer to add some enumerations like `slotPrimary, slotSecundary, againstPrimary, againSecondary` and so on, since we'll be working not only with `0` and `1` but also `2`, `3`... (yes, while they are still `0` and `1` for that particular slot set)

Reply:
Will they ever be other than zero and 1? At least the `slot` argument is always used as an array index. The against is against the 'primaries' and 'upgrades' fields (which reminds me that I need to rename those to 'primary' and 'upgrade' since they only refer to one thing.

It actually seems clearer to me to have these just be 0 and 1, although I can see how an enum makes it clearer. The 'slot' argument would end up being cast back to a usize anyway, though. Thoughts?